pkg/background/update_request_controller.go

```diff
@@ -370,6 +370,50 @@ func (c *controller) deletePolicy(obj interface{}) {
 	}
 }
 
+func (c *controller) deleteNSPolicy(obj interface{}) {
+	p, ok := kubeutils.GetObjectWithTombstone(obj).(*kyvernov1.Policy)
+	if !ok {
+		logger.Info("Failed to get deleted object", "obj", obj)
+		return
+	}
+
+	logger.V(4).Info("deleting policy", "name", p.Name)
+	key, err := cache.MetaNamespaceKeyFunc(kubeutils.GetObjectWithTombstone(obj))
+	if err != nil {
+		logger.Error(err, "failed to load policy key")
+	} else {
+		logger.V(4).Info("updating policy", "key", key)
+
+		// check if deleted policy is clone generate policy
+		generatePolicyWithClone := pkgCommon.ProcessDeletePolicyForCloneGenerateRule(p, c.client, c.kyvernoClient, c.urLister, p.GetName(), logger)
+
+		// get the generated resource name from update request
+		selector := labels.SelectorFromSet(labels.Set(map[string]string{
+			kyvernov1beta1.URGeneratePolicyLabel: p.Name,
+		}))
+
+		urList, err := c.urLister.List(selector)
+		if err != nil {
+			logger.Error(err, "failed to get ur for resource", "label", kyvernov1beta1.URGeneratePolicyLabel)
```

Comment:
change `"failed to get ur for resource"` this to `"failed to get UR for resource"`